src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if ([null, undefined].includes(this[i])) {
+        this[i] = '';
+      }
```

Comment:
You can also try to do this with one cycle
Just add this if statement to the next one, and if it's true, add only separator to the result string and use continue to move to the next cycle iteration